qiskit_ibm_provider/job/ibm_circuit_job.py

```diff
@@ -803,21 +652,31 @@ def _retrieve_result(self, refresh: bool = False) -> None:
 
         if not self._result or refresh:  # type: ignore[has-type]
             try:
-                result_response = self._api_client.job_result(
-                    self.job_id(), self._use_object_storage
-                )
-                self._set_result(result_response)
-                if self._status is JobStatus.ERROR:
-                    # Look for error message in result response.
-                    self._check_for_error_message(result_response)
+                api_result = self._runtime_client.job_results(self.job_id())
+                self._set_result(api_result)
             except ApiError as err:
                 if self._status not in (JobStatus.ERROR, JobStatus.CANCELLED):
                     raise IBMJobApiError(
                         "Unable to retrieve result for "
                         "job {}: {}".format(self.job_id(), str(err))
                     ) from err
 
-    def _set_result(self, raw_data: Optional[Dict]) -> None:
+    def _parse_result_for_errors(self, raw_data: str) -> str:
+        """Checks whether the job result contains errors
+
+        Args:
+            raw_data: Raw result data.
+
+        returns:
+            The error message, if found
+
+        """
+        result = re.search("JobError: '(.*)'", raw_data)
```

Comment:
Currently, if the regex search failed, we simply return the full job log and allow the user to look for the error. Of course, it can be nice to extend the regexp search to catch more errors, like you mention. Is there a comprehensive list?

Reply:
Maybe I'm not reading the code correctly, but it looks like it just [returns "unknown error"](https://github.com/Qiskit/qiskit-ibm-provider/blob/f3b5ac491e80d1a0997eeb99685450a3b2177114/qiskit_ibm_provider/job/ibm_circuit_job.py#L393-L400) if the regex fails? 
There is no comprehensive list because it's just whatever Python exception the program raises. Perhaps you can search for `Traceback` in the log instead?